tests/sources/test_mysql.py

```diff
@@ -667,11 +677,12 @@ async def test_validate_config_when_host_empty_then_raise_error():
 
 
 @pytest.mark.parametrize(
-    "tables_present_in_source, advanced_rules, expected_validation_result",
+    "tables_present_in_source, advanced_rules, id_in_source, expected_validation_result",
```

Comment:
Can you include tests to make sure that things still work if `id_in_source` is not provided? It looks like you modified all the pre-existing tests to specify column ids, but this change needs to remain backwards compatible.